subprojects/execution/src/main/java/org/gradle/internal/execution/history/impl/FingerprintMapSerializer.java

```diff
@@ -80,7 +82,9 @@ private HashCode readContentHash(FileType fileType, Decoder decoder) throws IOEx
             case Directory:
                 return FileSystemLocationFingerprint.DIR_SIGNATURE;
             case Missing:
-                return FileSystemLocationFingerprint.MISSING_FILE_SIGNATURE;
+                return MissingFileSnapshot.SIGNATURE;
+            case Unavailable:
+                return UnavailableFileSnapshot.SIGNATURE;
```

Comment:
This will never happen as `readFileType()` never returns `Unavailable`, nor does `writeFingerprint()` write it.

Reply:
Removed

Reply:
I'm guessing this will be reworked after today's discussion, so let's come back to it once you have the changes in.

Reply:
This file is unchanged in the latest version, is that what you expect to see here?

Reply:
@lptr this is why I've removed `case Unavailable:` from every case, since the tests only require https://github.com/gradle/gradle/pull/9746#discussion_r298253156